core/msd/src/main/rdl/Workload.tdl

```diff
@@ -4,24 +4,44 @@
 //Name types
 include "Names.tdl";
 
-// workload type describing workload associated with an identity
-type Workload Struct {
-    DomainName domainName; // name of the domain, optional for getWorkloadsByService API call
-    EntityName serviceName; // name of the service, , optional for getWorkloadsByService API call
+type StaticWorkloadType Enum {
+    VIP, // represents a virtual ip construct
+    ENTERPRISE_APPLIANCE, // represents generic network devices which can not be bootstrapped with an identity due to limitations
+    CLOUD_LB, // represents a public cloud load balancer ( for ingress to public cloud )
+    CLOUD_NAT, // represents a public cloud NAT gateway ( for egress from public cloud )
+    EXTERNAL_APPLIANCE // Appliance present outside of enterprise deployment locations. For SaaS / Third Party / Vendor use cases
+}
+
+
+// workload type describing workload bootstrapped with an identity
+type DynamicWorkload Struct {
+    DomainName domainName; // name of the domain
+    EntityName serviceName; // name of the service
     String uuid; // unique identifier for the workload, usually defined by provider
     Array<String> ipAddresses; // list of IP addresses associated with the workload, optional for getWorkloadsByIP API call
-    String hostname; //hostname associated with the workload
-    String provider; // infrastructure provider e.g. k8s, AWS, Azure, openstack etc.
+    String hostname; // hostname associated with the workload
+    String provider; // infrastructure provider e.g. Kubernetes, AWS, Azure, openstack etc.
     Timestamp updateTime; // most recent update timestamp in the backend
     Timestamp certExpiryTime; // certificate expiry time (ex: getNotAfter)
     Timestamp certIssueTime (optional); // certificate issue time (ex: getNotBefore)
 }
 
+// workload type describing workload indirectly associated with an identity ( without bootstrap )
+type StaticWorkload Struct {
+    DomainName domainName; // name of the domain
+    EntityName serviceName; // name of the service
+    StaticWorkloadType type; // value representing one of the StaticWorkloadType enum
+    Array<String> ipAddresses (optional); // list of IP addresses associated with the workload, optional for getWorkloadsByIP API call
+    String hostname (optional); // hostname associated with the workload
+    Timestamp updateTime; // most recent update timestamp in the backend
+}
+
 type WorkloadOptions Struct {
     Bool ipChanged; // boolean flag to signal a change in IP state
 }
 
 // list of workloads
 type Workloads Struct {
-    Array<Workload> workloadList; // list of workloads
+    Array<DynamicWorkload> dynamicWorkloadList; // list of dynamic workloads
+    Array<StaticWorkload> staticWorkloadList; // list of static workloads
```

Comment:
Since the Workloads object is changing anyways with Workload -> DynamicWorkload move, haven't kept staticWorkloadList as optional field.

Reply:
Is the expectation that the server will just pass back an empty array?

Reply:
Thats the behavior today with dynamic workloads ( empty array ). Would you prefer to make staticWorkloadList  optional instead?

Reply:
no, but there are a couple of places where we don't set it so we must need to make sure the server always returns an empty array.